### DISCUSSION TITLE
It seems that elasticsearch-py will have a function to deny connection to "Amazon Elasticsearch Service"

### DISCUSSION THREAD

**Discussion by nsano-rururu** [jertel/elastalert2#393] [Ideas]
Verify connection to Elasticsearch
https://github.com/elastic/elasticsearch-py/pull/1623

It seems that elasticsearch-py will have a function to deny connection to "Amazon Elasticsearch Service". Currently, elastalert2 supports connecting to Amazon Elasticsearch Service, but I can't connect.

AWS has announced that it will fork the client library in the version just before this check, and make it a new client library for the OpenSearch project.
https://aws.amazon.com/jp/blogs/opensource/keeping-clients-of-opensearch-and-elasticsearch-compatible-with-open-source/

What would you like to do?

**Comment by jertel** [comment]
It would be nice to have a config option that allows the user to choose which driver to use. Then default the option to which ever driver provides the most connection compatibility.

**Reply by nsano-rururu** [reply]
elastalert/create_index.py

```
from elasticsearch import RequestsHttpConnection
from elasticsearch.client import Elasticsearch
from elasticsearch.client import IndicesClient
from elasticsearch.exceptions import NotFoundError
```

```
elastalert/__init__.py
```

```
from elasticsearch import Elasticsearch
from elasticsearch import RequestsHttpConnection
from elasticsearch.client import _make_path
from elasticsearch.client import query_params
from elasticsearch.exceptions import TransportError
```

tests/base_test.py

```
from elasticsearch.exceptions import ConnectionError
from elasticsearch.exceptions import ElasticsearchException

```
elastalert/elastalert.py

```
from elasticsearch.exceptions import ConnectionError
from elasticsearch.exceptions import ElasticsearchException
from elasticsearch.exceptions import NotFoundError
from elasticsearch.exceptions import TransportError
```

**Reply by jertel** [reply]
We will need to isolate the ES clients into a dedicated ElastAlert layer. This will take time. As a temporary solution, is there a problem if we pin the client to the version that works with both? Or will newer ES servers reject the older clients?

**Reply by nsano-rururu** [reply]
I haven't confirmed the operation yet, so I'm just imagining, but since this change has been merged into the "7.x" branch, I think that it will be reflected in the past elasticsearch-py by backporting.
If it is not reflected, elasticsearch-py == 7.0.0, so there is no need to modify the program. Support is required when elasticsearch 8 is supported.

**Reply by nsano-rururu** [reply]
By the way, I don't use "Amazon Elasticsearch Service".

**Reply by nsano-rururu** [reply]
At least those who have already installed and used it should not be affected

**Reply by jertel** [reply]
I see now that our requirements.txt uses `== 7.0.0` not `>= 7.0.0`, and I also know that PyPi does not allow older versions to be re-used (or replaced). So yes, I agree, this project is not affected until we need to support 8.0.0. My suggestion is to hold off on upgrading the client to 8.x until there is a technical need to do so, and at that time we'll need to consider whether we should support users choosing which driver to utilize.

**Reply by nsano-rururu** [reply]
It's unlikely that elasticsearch 8 will be released soon, so I understand that it's okay for now.

**Reply by nsano-rururu** [reply]
By the way, Elasticsearch 8.0.0-alpha1 has been released.
